dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/AbstractHTTPDestination.java

```diff
@@ -299,25 +300,32 @@ protected void invokeComplete(final ServletContext context,
                                   final HttpServletRequest req,
                                   final HttpServletResponse resp,
                                   Message m) throws IOException {
-        ContinuationProvider p = m.get(ContinuationProvider.class);
+        //Liberty code change start
+        ContinuationProvider p = (ContinuationProvider) ((MessageImpl) m).getContinuationProvider();
+        //Liberty code change end
         if (p != null) {
             p.complete();
         }
     }
 
     private void copyKnownRequestAttributes(HttpServletRequest request, Message message) {
-        message.put(SERVICE_REDIRECTION, request.getAttribute(SERVICE_REDIRECTION));
+        //Liberty code change start
+        ((MessageImpl) message).setServiceRedirection(request.getAttribute(SERVICE_REDIRECTION));
+        //Liberty code change end
     }
 
-    protected void setupMessage(final Message inMessage,
+    //Liberty code change start
+    protected void setupMessage(final Message message,
```

Comment:
I might be missing something, but I don't think these changes (this line, line 323, lines 327-328) are necessary.  IIUC, the only time that the MessageImpl is used is on line 340 which is already casted to a MessageImpl - and already surrounded by the change tags.

Reply:
inMessage is used after the DelegatingInputStream within the method.